stabbity.lic

```diff
@@ -0,0 +1,253 @@
+=begin
+Stabbity, Stabbity, Stabbity. Repeat.
+=end
+custom_require.call(%w[common drinfomon equipmanager events])
+
+class Stabbity
+  include DRC
+
+  def initialize
+    unless DRStats.thief?
+      echo '*** But you\'re not a thief! ***'
+      exit
+    end
+    
+    arg_definitions = [
+      [
+        { name: 'mode', options: %w[arena single equip cleanup], optional: true, description: 'What mode to use' },
+        { name: 'noloot', options: %w[noloot], optional: true, description: 'Do not loot corpses' }
+      ]
+    ]
+
+    args = parse_args(arg_definitions)
+    @settings = get_settings
+    @debug = UserVars.stabbity_debug
+    @equipmanager = EquipmentManager.new
+    @preferred_weapon = @settings['stabbity']['weapons']['preferred']
+    @alternate_weapon = @settings['stabbity']['weapons']['alternate']
+    @thrown_weapon = @settings['stabbity']['weapons']['thrown']
+    @alternate_weapon_npcs = @settings['stabbity']['use_alternate_on']
+    @thrown_weapon_npcs = @settings['stabbity']['use_thrown_on']
+    @noloot = (args.mode == 'arena' || args.noloot) ? true : false
+    @mode = args.mode
+    @ignore_npcs = %w[Usdiwi Servant leopard construct zombie representative]
```

Comment:
We have an `ignored_npcs` setting that could be used here.

Reply:
Will use `ignored_npcs` setting in next release.